### DISCUSSION TITLE
how to calculate a buy/sell volume of a stock/future

### DISCUSSION THREAD

**Discussion by tiger2014** [DaveSkender/Stock.Indicators#960] [Ideas]
I want to know how to calculate a buy/sell volume of a stock/future. 

If we can calculate, we can decide how to do it easily.

Thanks.

**Comment by DaveSkender** [comment]
I'm not familiar with that technique, can you provide more explanation?

**Reply by tiger2014** [reply]
I have found out how to calculate it, we can use internal bars: 


`switch`  

        `close> open => upVol += volume ` 
        `close < open => dnVol -= volume`  
        
        `close > nz(close[1]) => upVol += volume`  
        `close < nz(close[1]) => dnVol -= volume ` 
       
        `nz(upVol[1]) > 0 => upVol += volume ` 
        `nz(dnVol[1]) < 0 => dnVol -= volume  `

**Reply by DaveSkender** [reply]
This looks like an up/down split version of [On-balance Volume (OBV)](https://dotnet.stockindicators.dev/indicators/Obv/#content)
Do you have any reputable author references or common name for this indicator?
I'm curious to see if this is just extra info we can add to OBV output or as a new indicator.

**Reply by tiger2014** [reply]
I got it from this link: [https://www.tradingview.com/chart/?solution=43000672561](https://www.tradingview.com/chart/?solution=43000672561)

**Reply by DaveSkender** [reply]
Maybe Gregory Morris is an authoritative source for these [up/down family](https://stockcharts.com/articles/dancing/2015/11/up-volume-down-volume-indicators-chap-8--cgmbi.html) of indicators, but maybe also [Sherman and Marian McClellan](https://www.mcoscillator.com/about_us/)
